preferred-operator/interface.js

```diff
@@ -0,0 +1,366 @@
+      // Set the priority to medium
+      priorities.set(selectedOperator.dataset.id, {
+        priority: 'medium',
+        color: 'priority',
+        icon: 'light-medium-priority',
+        label: 'Medium priority',
+      });
+      break;
+    case 3:
+      // Set the priority to low
+      priorities.set(selectedOperator.dataset.id, {
+        priority: 'low',
+        color: 'priority',
+        icon: 'light-low-priority',
+        label: 'Low priority',
+      });
+      break;
+    default:
+      // Set the priority to excluded
+      priorities.set(selectedOperator.dataset.id, {
+        priority: 'excluded',
+        color: 'excluded',
+        icon: 'light-exclude-priority',
+        label: 'Excluded priority',
+      });
+      break;
+  }
+
+  // When we changed the priority we let the UI reflect it by calling this function
+  setPriorityStyling();
+};
+
+/**
+ * Sets the correct styling based on the priority that was selected inside the priority menu.
+ */
+const setPriorityStyling = () => {
+  const priority = priorities.get(selectedOperator.dataset.id);
+
+  /* eslint-disable */
+  selectedOperator.querySelector('.priority-list-icon').className = `priority-list-icon ${
+    priority ? priority.color : ''
+  }`;
+  selectedOperator.querySelector('.priority-list-icon img').src = `images/${
+    priority ? priority.icon : 'light-no-priority'
+  }.svg`;
+  selectedOperator.querySelector('.priority-label').innerHTML = `${priority ? priority.label : 'No priority'}`;
+  /* eslint-enable */
+};
+
+/**
+ * This recalculates the route based upon the new operator selection.
+ * It extracts the priorities that were set in the list and formats them to how our API is expecting the data
+ * If there are no priorities it just calculates a regular route.
+ */
+const recalculateRoute = () => {
+  loadingToast.style.transform = `translateY(0)`;
+
+  if (priorities.size > 0) {
+    const level1 = [];
+    const level2 = [];
+    const level3 = [];
+    const exclude = [];
+
+    // eslint-disable-next-line no-unused-vars
+    for (const [id, { priority }] of priorities.entries()) {
+      switch (priority) {
+        case 'high':
+          level1.push(id);
+          break;
+        case 'medium':
+          level2.push(id);
+          break;
+        case 'low':
+          level3.push(id);
+          break;
+        case 'excluded':
+          exclude.push(id);
+          break;
+        default:
+          break;
+      }
+    }
+
+    /**
+     * So by default our operator preference is set to none.
+     * Whenever you add a preferred operator you will have to tell our API how you would like to use them; preferred or required.
+     * Preferred will prefer the operator ranking when calculating routes
+     * Required will require the operator ranking when calculating routes, excluded operators will be ignored
+     *
+     * Our example only uses level 1 to 3. Our API has 10 levels available for if you want to be more specific.
```

Comment:
Add:

'Using a lower level (such as level 7) for the lowest priority makes it less likely for the operator to be used in the documentation.'

Reply:
Added + link to our documentation.